pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -287,6 +287,27 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 		apiEndpoint, serviceAccountToken, caCert, err = p.driverUpdate(cluster, *spec)
 	} else {
 		logrus.Infof("Updating cluster [%s]", cluster.Name)
+
+		updating := false
+		for i := 0; i < 4; i++ {
+			if cluster, err = p.Clusters.Get(cluster.Name, metav1.GetOptions{}); err != nil {
+				logrus.Infof("Failed to get cluster, retrying")
+				time.Sleep(1 * time.Second)
+				continue
+			}
+			v3.ClusterConditionUpdated.Unknown(cluster)
+			if _, err = p.Clusters.Update(cluster); err != nil {
```

Comment:
continue only if apierrors.IsConflict 

can refer to this example if required, https://github.com/rancher/rancher/blob/master/pkg/controllers/management/multiclusterapp/controller_multiclusterapp.go#L283